core/src/main/java/dk/alexandra/fresco/framework/util/RegularBitVector.java

```diff
@@ -0,0 +1,129 @@
+package dk.alexandra.fresco.framework.util;
+
+import java.util.BitSet;
+
+/**
+ * Class for representing a vector of bits. Uses {@link BitSet} to hold the vector.
+ * This class allows the construction of bit vector containing an arbitrary amount
+ * of elements. Specifically the amount of bits does not need to be divisible by 8.
+ */
+public class RegularBitVector implements BitVector {
+
+  private final BitSet bits;
+  private final int size;
+
+  /**
+   * Creates a BitVector from a <code>boolean</code> array.
+   *
+   * @param array
+   *          a <code>boolean</code> array
+   */
+  public RegularBitVector(boolean[] array) {
+    this.bits = BitSetUtils.fromArray(array);
+    this.size = array.length;
+  }
+
+  /**
+   * Creates a BitVector from an array of bytes.
+   *
+   * @param array an array of bytes
+   * @param size the number of bits of <code>array</code> to use for this vector
+   */
+  public RegularBitVector(byte[] array, int size) {
+    if (size < 0) {
+      throw new IllegalArgumentException(
+          "Size of vector must not be negative but was " + size);
+    }
+    this.bits = BitSet.valueOf(array);
+    this.size = size;
+  }
+
+  /**
+   * Creates a BitVector with all entries set to zero.
+   *
+   * @param size size of the vector
+   */
+  public RegularBitVector(int size) {
+    if (size < 0) {
```

Comment:
There is no reason for this check, BitSet already performs it

Reply:
Check removed in commit dddb1730